pkg/build/builder/sti.go

```diff
@@ -283,8 +297,17 @@ func (s *S2IBuilder) Build() error {
 		handleBuildStatusUpdate(s.build, s.client, nil)
 		return err
 	}
+	for _, stage := range result.BuildInfo.Stages {
+		for _, step := range stage.Steps {
+			if s.build.Status.Stages, err = api.RecordStageAndStepInfo(s.build.Status.Stages, api.StageName(stage.Name), api.StepName(step.Name), unversioned.NewTime(step.StartTime), unversioned.NewTime(step.StartTime.Add(time.Duration(step.DurationMilliseconds)*time.Millisecond))); err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
The stages/steps returned from the s2i code is the wrong type.  I guess I could just cast everything in the merge function.